File: ecma6/euler010.js
/**
 * Created by gcharles on 4/19/16.
 */
/**
 The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

 Find the sum of all the primes below two million.
*/
var euler010 = new EulerProblem({
   problem: 10,
   testInput: 10,
   testOutput: 17,
   realInput: 2000000,
   realOutput: 142913828922,

   solver: function(n) {

      return gwu.reduce((sum, x) => sum + x,
             gwu.takeWhile(p => p < n,
             primes.PrimeGenerator())
      );
   }
});

File: python/euler010.py
"""
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
"""
from itertools import takewhile

from common.euler_lib import prime_generator
from euler import euler_problem


@euler_problem()
def euler010(n: int | str) -> int:
    return sum(takewhile(lambda p: p < int(n), prime_generator()))


if __name__ == '__main__':
    print(euler010(10))
    print(euler010(2_000_000))
